### ISSUE TITLE
Error: Could not fetch client credentials token for service of type connectivity: XsuaaService requires uaadomain inside credentials

### ISSUE THREAD

**Issue by gauravkr75** [SAP/cloud-sdk-js#4841]
**Describe the bug**
We are using SAP Cloud SDK with CAP.
When we are trying to call an on-premise service using cloud-sdk, we are getting the error: `Error: Could not fetch client credentials token for service of type connectivity: XsuaaService requires uaadomain inside credentials`.
This issue started coming since we upgraded the cloud sdk from version `3.15.0` to `3.16.0`

It seems the issue is caused because the connectivity service credentials does not contain `uaadomain` property and these credentials are being passed to the `XsuaaService` class of `@sap/xssec` npm package which looks for this property and then throws an error. Please check the **Screenshots** section to see this flow.

**To Reproduce**

1. Set up ...
2. Execute ...
3. Confirm ...

**Expected behavior**
Receive the response without any errors

**Screenshots**
This screenshot explains the process flow and where it fails.

![image](https://github.com/user-attachments/assets/197a8497-2e16-44fe-abab-7bfa859602d7)

**Used Versions:**

- node version via `node -v`: `v20.12.2`
- npm version via `npm -v`: `10.5.0`
- SAP Cloud SDK version you used as dependency: `3.16.0`

**Code Examples**
```js
import { executeHttpRequest } from "@sap-cloud-sdk/http-client";
const response = await executeHttpRequest(
            { destinationName: "destination-name" },
            {
                method, url, data, headers
            }
        );
```

**Log file**
If applicable, add your log file or related error message. Again, please remove your sensitive information.

Full Log:
```
[cds] - ErrorWithCause: Failed to load destination.
    at /Users/<USER>/Documents/project/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/destination/destination-accessor.ts:67:13
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async resolveDestination (/Users/<USER>/Documents/project/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/destination/destination-accessor.ts:65:31)
    at async /Users/<USER>/Documents/project/node_modules/@sap-cloud-sdk/http-client/src/http-client.ts:83:33
Caused by:
ErrorWithCause: Failed to add proxy authorization header - client credentials grant failed!
    at proxyAuthorizationHeader (/Users/<USER>/Documents/project/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/connectivity-service.ts:123:11)
    ... 21 lines matching cause stack trace ...
    at processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: ErrorWithCause: Failed to add proxy authorization header - client credentials grant failed!
      at proxyAuthorizationHeader (/Users/<USER>/Documents/project/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/connectivity-service.ts:123:11)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async proxyHeaders (/Users/<USER>/Documents/project/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/connectivity-service.ts:102:27)
      ... 5 lines matching cause stack trace ...
  Caused by:
  ErrorWithCause: Could not fetch client credentials token for service of type "connectivity".
      at serviceToken (/Users/<USER>/Documents/project/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/token-accessor.ts:70:11)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async proxyAuthorizationHeader (/Users/<USER>/Documents/project/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/connectivity-service.ts:120:19)
      at async proxyHeaders (/Users/<USER>/Documents/project/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/connectivity-service.ts:102:27)
      at async addProxyConfigurationOnPrem (/Users/<USER>/Documents/project/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/connectivity-service.ts:41:11)
      at async Function.getDestinationFromDestinationService (/Users/<USER>/Documents/project/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/destination/destination-from-service.ts:158:30)
      at async getDestination (/Users/<USER>/Documents/project/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/destination/destination-accessor.ts:98:6)
      at async resolveDestination (/Users/<USER>/Documents/project/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/destination/destination-accessor.ts:65:31)
      at async /Users/<USER>/Documents/project/node_modules/@sap-cloud-sdk/http-client/src/http-client.ts:83:33
  Caused by:
  Error: Could not fetch client credentials token for service of type connectivity: XsuaaService requires uaadomain inside credentials.
      at /Users/<USER>/Documents/project/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/xsuaa-service.ts:60:11
      at processTicksAndRejections (node:internal/process/task_queues:95:5) {
    cause: ErrorWithCause: Could not fetch client credentials token for service of type "connectivity".
        at serviceToken (/Users/<USER>/Documents/project/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/token-accessor.ts:70:11)
        at processTicksAndRejections (node:internal/process/task_queues:95:5)
        at async proxyAuthorizationHeader (/Users/<USER>/Documents/project/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/connectivity-service.ts:120:19)
        at async proxyHeaders (/Users/<USER>/Documents/project/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/connectivity-service.ts:102:27)
        at async addProxyConfigurationOnPrem (/Users/<USER>/Documents/project/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/connectivity-service.ts:41:11)
        at async Function.getDestinationFromDestinationService (/Users/<USER>/Documents/project/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/destination/destination-from-service.ts:158:30)
        at async getDestination (/Users/<USER>/Documents/project/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/destination/destination-accessor.ts:98:6)
        at async resolveDestination (/Users/<USER>/Documents/project/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/destination/destination-accessor.ts:65:31)
        at async /Users/<USER>/Documents/project/node_modules/@sap-cloud-sdk/http-client/src/http-client.ts:83:33
    Caused by:
    Error: Could not fetch client credentials token for service of type connectivity: XsuaaService requires uaadomain inside credentials.
        at /Users/<USER>/Documents/project/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/xsuaa-service.ts:60:11
        at processTicksAndRejections (node:internal/process/task_queues:95:5) {
      cause: Error: Could not fetch client credentials token for service of type connectivity: XsuaaService requires uaadomain inside credentials.
          at /Users/<USER>/Documents/project/node_modules/@sap-cloud-sdk/connectivity/src/scp-cf/xsuaa-service.ts:60:11
          at processTicksAndRejections (node:internal/process/task_queues:95:5)
    }
  }
```

**Impact / Priority**
For now we have downgraded the version to `3.15.0` so its working but it would be great if it can be fixed soon.

Affected development phase: e.g. Getting Started, Development, Release, Production

Impact: e.g. No Impact, Inconvenience, Impaired, Blocked

Timeline: e.g. Go-Live is in 12 weeks.

**Additional context**

**Comment by deekshas8** [comment]
Hi @gauravkr75 ,

Thanks you for raising this. We are aware of this issue in the xssec library and it is currently being worked on by the team. Please use 3.15.0 for now.
@finkmanAtSap @jjtang1985

**Comment by deekshas8** [comment]
Hi @gauravkr75 ,

We released a canary version with the fix. You can try it out setting this in your package.json:
`"@sap-cloud-sdk/connectivity": "canary"`.

We will likely release a new minor version sometime next week.

**Comment by gauravkr75** [comment]
Hi @deekshas8,

Thanks for looking into this issue. I tested with the canary version and it is working fine now.

**Comment by deekshas8** [comment]
Hi @gauravkr75 

We have now released a new minor v3.17.0.
